packages/server/src/fhir/references.ts

```diff
@@ -79,11 +84,18 @@ export class FhirReferenceValidator<T extends Resource> {
     if (refParts.length !== 2) {
       return;
     }
+    if (!this.root.meta?.project) {
+      return;
+    }
 
     try {
       const systemRepo = getSystemRepo();
-      const existing = await systemRepo.readReference(reference);
-      if (existing.meta?.project && this.root.meta?.project && existing.meta.project !== this.root.meta.project) {
+      const target = await systemRepo.readReference(reference);
```

Comment:
(non-blocking)

Out of scope for this PR, but I'm realizing the inefficiencies of the original implementation of this file.

Currently it's doing the naive thing, crawling the resource, and synchronously checking the references.

The server `Repository` class already has a `readReferences(references: Reference[])` method.  We could easily modify this file to first gather all the references, and then do it as a batch operation.

The implementation of `readReferences` uses a batch Redis operation.

The implementation of `readReferences`  could itself be optimized to read multiple references in a single SQL query.  That is more complicated though.

Reply:
That sounds good to me: I can knock the efficiency boost out in a fast follow PR